### PR TITLE
nodejs-{17,16,14,12}_x: security updates

### PR DESCRIPTION
**Pull request by mohe2015**
###### Motivation for this change
https://nodejs.org/en/blog/vulnerability/jan-2022-security-releases/
CVE-2021-44531
CVE-2021-44532
CVE-2021-44533
CVE-2022-21824

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mohe2015** [comment]
sorry for the bad format

<summary>
<details>
Link to currently reviewing PR:                                                                                                                             
https://github.com/NixOS/nixpkgs/pull/154474                                                                                                                
                                                                                                                                                            
138 packages marked as broken and skipped:                                                                                                                  
airfield cq-editor falkon foo-yc20 hyperspace-cli kexi libsForQt512.akonadi-calendar libsForQt512.akonadi-calendar-tools libsForQt512.akonadi-contacts libsF
orQt512.akonadi-import-wizard libsForQt512.akonadiconsole libsForQt512.akregator libsForQt512.angelfish libsForQt512.bismuth libsForQt512.calendarsupport li
bsForQt512.eventviews libsForQt512.incidenceeditor libsForQt512.kaccounts-providers libsForQt512.kaddressbook libsForQt512.kalarm libsForQt512.kde-cli-tools
 libsForQt512.kdepim-addons libsForQt512.kdepim-runtime libsForQt512.kdeplasma-addons libsForQt512.kgpg libsForQt512.khotkeys libsForQt512.kinfocenter libsF
orQt512.kio-gdrive libsForQt512.kmail libsForQt512.kmail-account-wizard libsForQt512.kmailtransport libsForQt512.kmenuedit libsForQt512.knotes libsForQt512.
konqueror libsForQt512.kontact libsForQt512.korganizer libsForQt512.kreport libsForQt512.krohnkite libsForQt512.ksystemstats libsForQt512.kwin-dynamic-works
paces libsForQt512.kwin-tiling libsForQt512.libgravatar libsForQt512.libkdepim libsForQt512.libkgapi libsForQt512.libksieve libsForQt512.libksysguard libsFo
rQt512.mailcommon libsForQt512.mailimporter libsForQt512.marble libsForQt512.mbox-importer libsForQt512.messagelib libsForQt512.parachute libsForQt512.pim-d
ata-exporter libsForQt512.pim-sieve-editor libsForQt512.pimcommon libsForQt512.plasma-browser-integration libsForQt512.plasma-desktop libsForQt512.plasma-di
sks libsForQt512.plasma-phone-components libsForQt512.plasma-systemmonitor libsForQt512.plasma-vault libsForQt512.plasma-workspace libsForQt512.powerdevil l
ibsForQt512.systemsettings libsForQt514.akonadi-calendar libsForQt514.akonadi-calendar-tools libsForQt514.akonadi-contacts libsForQt514.akonadi-import-wizar
d libsForQt514.akonadiconsole libsForQt514.akregator libsForQt514.angelfish libsForQt514.bismuth libsForQt514.calendarsupport libsForQt514.eventviews libsFo
rQt514.incidenceeditor libsForQt514.kaccounts-providers libsForQt514.kaddressbook libsForQt514.kalarm libsForQt514.kde-cli-tools libsForQt514.kdepim-addons 
libsForQt514.kdepim-runtime libsForQt514.kdeplasma-addons libsForQt514.kgpg libsForQt514.khotkeys libsForQt514.kinfocenter libsForQt514.kio-gdrive libsForQt
514.kmail libsForQt514.kmail-account-wizard libsForQt514.kmailtransport libsForQt514.kmenuedit libsForQt514.knotes libsForQt514.konqueror libsForQt514.konta
ct libsForQt514.korganizer libsForQt514.kreport libsForQt514.krohnkite libsForQt514.ksystemstats libsForQt514.kwin-dynamic-workspaces libsForQt514.kwin-tili
ng libsForQt514.libgravatar libsForQt514.libkdepim libsForQt514.libkgapi libsForQt514.libksieve libsForQt514.libksysguard libsForQt514.mailcommon libsForQt5
14.mailimporter libsForQt514.marble libsForQt514.mbox-importer libsForQt514.messagelib libsForQt514.parachute libsForQt514.pim-data-exporter libsForQt514.pi
m-sieve-editor libsForQt514.pimcommon libsForQt514.plasma-browser-integration libsForQt514.plasma-desktop libsForQt514.plasma-disks libsForQt514.plasma-phon
e-components libsForQt514.plasma-systemmonitor libsForQt514.plasma-vault libsForQt514.plasma-workspace libsForQt514.powerdevil libsForQt514.systemsettings m
ultibootusb purePackages.faust python310Packages.dask-xgboost python310Packages.optuna python310Packages.rl-coach python39Packages.arviz python39Packages.da
sk-xgboost python39Packages.optuna python39Packages.pymc3 python39Packages.rl-coach remarkjs teleprompter timedoctor toggldesktop vimPlugins.coc-imselect wh
irlpool-gui                                                                                                                                                 
                                                                                                                                                            
1 package blacklisted:                                                                                                                                      
tests.writers

236 packages failed to build:                                                                                                                       [58/972]
amarok anki ankisyncd apache-airflow calamares calibre calligra chromium clipgrab code-server csound-qt cutter cwltool digikam eagle fast-cli faustStk fcitx
-engines.libpinyin fcitx5-chinese-addons foxdot freecad frescobaldi fx_cast_bridge ghostwriter globalprotect-openconnect golden-cheetah hqplayer-desktop hyd
rus image_optim insync-v3 isso jami-client-qt jellyfin-media-player jellyfin-mpv-shim kalendar kbibtex kdeltachat kdev-php kdev-python kdevelop kdevelop-unw
rapped kiwix lemmy-ui libsForQt5.akonadi-calendar libsForQt5.akonadi-calendar-tools libsForQt5.akonadi-contacts libsForQt5.akonadi-import-wizard libsForQt5.
akonadiconsole libsForQt5.akregator libsForQt5.angelfish libsForQt5.bismuth libsForQt5.calendarsupport libsForQt5.eventviews qt5Full libsForQt5.incidenceedi
tor libsForQt5.kaccounts-providers libsForQt5.kaddressbook libsForQt5.kalarm libsForQt5.kde-cli-tools libsForQt5.kdepim-addons libsForQt5.kdepim-runtime lib
sForQt5.kdeplasma-addons libsForQt5.kgpg libsForQt5.khotkeys libsForQt5.kinfocenter libsForQt5.kio-gdrive libsForQt5.kmail libsForQt5.kmail-account-wizard l
ibsForQt5.kmailtransport libsForQt5.kmenuedit libsForQt5.knotes libsForQt5.konqueror libsForQt5.kontact libsForQt5.korganizer libsForQt5.kreport libsForQt5.
krohnkite libsForQt5.ksystemstats libsForQt5.kwin-dynamic-workspaces libsForQt5.kwin-tiling libsForQt5.libgravatar libsForQt5.libkdepim libsForQt5.libkgapi 
libsForQt5.libksieve libsForQt5.libksysguard libsForQt5.mailcommon libsForQt5.mailimporter libsForQt5.marble libsForQt5.mbox-importer libsForQt5.messagelib 
libsForQt5.parachute libsForQt5.pim-data-exporter libsForQt5.pim-sieve-editor libsForQt5.pimcommon libsForQt5.plasma-browser-integration libsForQt5.plasma-d
esktop libsForQt5.plasma-disks libsForQt5.plasma-phone-components libsForQt5.plasma-systemmonitor libsForQt5.plasma-vault libsForQt5.plasma-workspace libsFo
rQt5.powerdevil qt5.qtwebengine qt5.qtwebview libsForQt5.systemsettings qt512.full qt512.qtwebengine qt512.qtwebview qt514.full qt514.qtwebengine qt514.qtwe
bview luminanceHDR magnetophonDSP.ConstantDetuneChorus mellowplayer mendeley merkaartor mindforger minizincide mnemosyne musescore mx-puppet-discord n8n nap
ari nextcloud-client nmapsi4 oni2 onionshare-gui p4v patray peertube pentobi plex-media-player psi psitransfer puppeteer-cli python-qt python310Packages.apa
che-airflow python310Packages.aplpy python310Packages.asdf python310Packages.batchgenerators python310Packages.batchspawner python310Packages.caffe python31
0Packages.dask-gateway python310Packages.dask-image python310Packages.dask-jobqueue python310Packages.dask-mpi python310Packages.distributed python310Packag
es.dockerspawner python310Packages.foxdot python310Packages.glymur python310Packages.image-match python310Packages.imagecorruptions python310Packages.imgaug
 python310Packages.intensity-normalization python310Packages.jupyterhub python310Packages.jupyterhub-ldapauthenticator python310Packages.jupyterhub-systemds
pawner python310Packages.jupyterhub-tmpauthenticator python310Packages.livelossplot python310Packages.magicgui python310Packages.mask-rcnn python310Packages
.napari python310Packages.nbsmoke python310Packages.oauthenticator python310Packages.pims python310Packages.pyqtwebengine python310Packages.pyregion python3
10Packages.pyside2 python310Packages.pyside2-tools python310Packages.pytest-astropy python310Packages.pytest-astropy-header python310Packages.pywebview pyth
on310Packages.reproject python310Packages.scikitimage python310Packages.slicedimage python310Packages.spectral-cube python310Packages.streamz python310Packa
ges.sunpy python310Packages.tensorflow-datasets python310Packages.tumpa python310Packages.wandb python39Packages.batchspawner python39Packages.dockerspawner
 python39Packages.jupyterhub python39Packages.jupyterhub-ldapauthenticator python39Packages.jupyterhub-systemdspawner python39Packages.jupyterhub-tmpauthent
icator python39Packages.magicgui python39Packages.oauthenticator python39Packages.pyqtwebengine python39Packages.pyside2 python39Packages.pyside2-tools pyth
on39Packages.pywebview python39Packages.qiskit python39Packages.tensorly python39Packages.tumpa qmapshack qolibri qsyncthingtray qutebrowser reveal-md rssgu
ard rstudio rstudioWrapper schildichat-desktop schildichat-desktop-wayland schildichat-web seafile-client semantik sharedown sigil skrooge sl1-to-photon son
ic-pi sourcehut.metasrht stremio supercollider supercollider_scel syncplay syncthingtray tageditor teamspeak_client teamviewer tellico texmaker trojita uivo
nim vnote wacomtablet webmacs whitebophir yacas-gui yuzu-ea yuzu-mainline zanshin zeal zombietrackergps

280 packages built:
adoptopenjdk-icedtea-web antennas antora balanceofsatoshis bat-extras.prettybat bitwarden-cli botamusique bs-platform castnow cinnamon.xreader cjdns cjdns-t
ools commitlint concurrently create-cycle-app cryptpad csslint cz-cli deltachat-desktop discourse discourseAllPlugins element-desktop element-desktop-waylan
d elmPackages.create-elm-app elmPackages.elm elmPackages.elm-analyse elmPackages.elm-coverage elmPackages.elm-doc-preview elmPackages.elm-git-install elmPac
kages.elm-language-server elmPackages.elm-live elmPackages.elm-optimize-level-2 elmPackages.elm-review elmPackages.elm-test elmPackages.elm-upgrade elmPacka
ges.elm-verify-examples elmPackages.elm-xref emojione emscripten epgstation faust faust2alqt faust2alsa faust2csound faust2firefox faust2jack faust2jackrust
 faust2jaqt faust2ladspa faust2lv2 faustPhysicalModeling faustlive firefox-esr-unwrapped xulrunner fixup_yarn_lock flexoptix-app flood fx github-runner gitl
ab gitlab-ee gnomeExtensions.pop-shell gotify-server grafana-image-renderer gtop hedgedoc heroku hueadm imapnotify imgbrd-grabber intensity-normalization io
sevka jellyfin jellyfin-web jiten joplin kapitonov-plugins-pack kibana7 ldgallery lessc lumo magnetophonDSP.CharacterCompressor magnetophonDSP.CompBus magne
tophonDSP.LazyLimiter magnetophonDSP.MBdistortion magnetophonDSP.RhythmDelay magnetophonDSP.VoiceOfFaust magnetophonDSP.faustCompressors magnetophonDSP.plug
inUtils magnetophonDSP.shelfMultiBand malt mastodon mastodon-bot matrix-alertmanager matrix-appservice-discord matrix-appservice-irc matrix-appservice-slack
 matrix-recorder mbqn meshcentral mirakurun mjolnir mkchromecast mooSpace netlify-cli newman nodejs nodejs-12_x nodejs-14_x nodejs-17_x nodejs-slim nodejs-s
lim-12_x nodejs-slim-14_x nodejs-slim-17_x odoo onlykey opa openmoji-black openmoji-color openvscode-server parity-ui plausible pm2 postcss-cli powerdns-adm
in prefetch-yarn-deps prometheus pscid pulp purescript-psa pyright python310Packages.bokeh python310Packages.casa-formats-io python310Packages.dask python31
0Packages.dask-glm python310Packages.holoviews python310Packages.hvplot python310Packages.panel python310Packages.pyfftw python39Packages.aplpy python39Pack
ages.asdf python39Packages.batchgenerators python39Packages.bokeh python39Packages.caffe python39Packages.casa-formats-io python39Packages.clifford python39
Packages.dask python39Packages.dask-gateway python39Packages.dask-glm python39Packages.dask-image python39Packages.dask-jobqueue python39Packages.dask-ml py
thon39Packages.dask-mpi python39Packages.datashader python39Packages.distributed python39Packages.glymur python39Packages.holoviews python39Packages.hvplot 
python39Packages.image-match python39Packages.imagecorruptions python39Packages.imgaug python39Packages.intake python39Packages.intake-parquet python39Packa
ges.livelossplot python39Packages.mask-rcnn python39Packages.nbsmoke python39Packages.panel python39Packages.pims python39Packages.pyfftw python39Packages.p
yregion python39Packages.pytest-astropy python39Packages.pytest-astropy-header python39Packages.qiskit-machine-learning python39Packages.reproject python39P
ackages.scikitimage python39Packages.slicedimage python39Packages.sparse python39Packages.spectral-cube python39Packages.streamz python39Packages.stumpy pyt
hon39Packages.stytra python39Packages.sunpy python39Packages.tensorflow-datasets python39Packages.wandb redoc-cli sage sageWithDoc shepherd shout slack slim
erjs sloc sourcehut.buildsrht sourcehut.coresrht sourcehut.dispatchsrht sourcehut.gitsrht sourcehut.hgsrht sourcehut.hubsrht sourcehut.listssrht sourcehut.m
ansrht sourcehut.pastesrht sourcehut.todosrht spacegun styx tambura teams teck-programmer thelounge thunderbird-unwrapped triton uchiwa ungoogled-chromium u
nity3d unityhub vieb vimPlugins.YouCompleteMe vimPlugins.coc-clangd vimPlugins.coc-cmake vimPlugins.coc-css vimPlugins.coc-diagnostic vimPlugins.coc-emmet v
imPlugins.coc-eslint vimPlugins.coc-explorer vimPlugins.coc-git vimPlugins.coc-go vimPlugins.coc-highlight vimPlugins.coc-html vimPlugins.coc-java vimPlugin
s.coc-jest vimPlugins.coc-json vimPlugins.coc-lists vimPlugins.coc-markdownlint vimPlugins.coc-metals vimPlugins.coc-pairs vimPlugins.coc-prettier vimPlugin
s.coc-pyright vimPlugins.coc-python vimPlugins.coc-r-lsp vimPlugins.coc-rls vimPlugins.coc-rust-analyzer vimPlugins.coc-smartf vimPlugins.coc-snippets vimPl
ugins.coc-solargraph vimPlugins.coc-stylelint vimPlugins.coc-tabnine vimPlugins.coc-texlab vimPlugins.coc-tslint vimPlugins.coc-tslint-plugin vimPlugins.coc
-tsserver vimPlugins.coc-ultisnips vimPlugins.coc-vetur vimPlugins.coc-vimlsp vimPlugins.coc-vimtex vimPlugins.coc-wxml vimPlugins.coc-yaml vimPlugins.coc-y
ank vimPlugins.markdown-preview-nvim vscode vscode-extensions.matklad.rust-analyzer vscode-extensions.ms-python.vscode-pylance vscode-extensions.vadimcn.vsc
ode-lldb vscode-fhs vscode-with-extensions vscodium vscodium-fhs wasm-bindgen-cli wasm-strip wasm-text-gen wast-refmt webassemblyjs-cli webassemblyjs-repl w
halebird wring yaml-language-server yarn yarn2nix ycmd zerobin zigbee2mqtt zx
</details>
</summary>

**Review by mohe2015 — COMMENTED** [review]
`
nix-build -A nixosTests.peertube -A nixosTests.matrix-appservice-irc -A nixosTests.gitlab -A nixosTests.firefox 
`
(just quickly chosen based on rebuilds)

Edit: I f*ed up - wrong branch

Edit2: Also seems to work on this branch

**Comment by mweinelt** [comment]
Did a lot of rebuilds and breakages looked unrelated and where mostly elsewhere, e.g. in python310.

**Comment by mohe2015** [comment]
Should we bump the release branch to latest nodejs versions as there are not published version bumps for previous minor versions? Or should be patch?
